packages/autorest.python/autorest/codegen/templates/serialization.py.jinja2

```diff
@@ -664,8 +664,9 @@ class Serializer(object):
                                 _serialized.update(_new_attr)  # type: ignore
                             _new_attr = _new_attr[k]  # type: ignore
                             _serialized = _serialized[k]
-                except ValueError:
-                    continue
+                except ValueError as err:
+                    if isinstance(err, SerializationError):
+                        raise err
```

Comment:
To pass test and keep same with msrest: https://github.com/Azure/msrest-for-python/blob/af41991f17445e4cb26d39c247f091ddfd7027b9/msrest/serialization.py#L572-L575

Reply:
Yes we should do that. Should have been just `raise` though:

```suggestion
                        raise
```

Reply:
![image](https://github.com/Azure/autorest.python/assets/70930885/bc097263-f523-4d41-ae36-d76f0ab7a15e)

![image](https://github.com/Azure/autorest.python/assets/70930885/90c41b29-29a5-41d2-a1c5-73959a129dd3)

Got it. Direct `raise` have cleaner traceback